plugins/woocommerce-blocks/assets/js/data/cart/index.ts

```diff
@@ -27,10 +38,29 @@ const store = createReduxStore( STORE_KEY, {
 	controls: dataControls,
 	selectors,
 	resolvers,
+	initialState: {
+		...defaultCartState,
+		cartData: {
+			...defaultCartState.cartData,
+			...( persistenceLayer.get() || {} ),
+		},
+	},
 } );
 
 register( store );
 
+// The resolver for getCartData fires off an API request. But if we know the cart is empty, we can skip the request.
+// Likewise, if we have a valid persistent cart, we can skip the request.
+// The only reliable way to check if the cart is empty is to check the cookies.
+window.addEventListener( 'load', () => {
+	if (
+		( ! hasCartSession() || persistenceLayer.get() ) &&
+		! isAddingToCart
```

Comment:
I added a check for isAddingToCart because on those requests we're most likely going to need to do an API request, and also the playwright test suite was expecting a request.